### PR TITLE
users and customers-apis

### PR DESCRIPTION
**Pull request by EstherFrauvirt**

### CODE REVIEW

File: src/main/java/com/yellow/ordermanageryellow/Service/UsersService.java [hunk 1]
```diff
@@ -1,40 +1,109 @@
...
+            return UserRepository.save(newUser);
+        } else
+            throw new ObjectExistException("user is already exist");
     }
-}
-    public String generateToken(Users user){
-        return user.getCompanyId() + user.getId()+ user.getRoleId();
+
+    public String generateToken(Users user) {
+        return user.getCompanyId() + user.getId() + user.getRoleId();
```
Comment (moriyahazan) [inline src/main/java/com/yellow/ordermanageryellow/Service/UsersService.java:60]: please contact with token like & :  company +&  + user + & + role

File: src/main/java/com/yellow/ordermanageryellow/Service/UsersService.java [hunk 2]
```diff
@@ -1,40 +1,109 @@
...
+        for (Users user : users) {
+            userMap.put(user.getId(), user.getFullName());
+        }
+        return userMap;
+    }
+
+    @SneakyThrows
+    public List<UserDTO> getCustomers(int pageNumber, int pageSize) {
+        Pageable pageable = PageRequest.of(pageNumber, pageSize);
+        //company and role will be taken from token
```
Comment (moriyahazan) [inline src/main/java/com/yellow/ordermanageryellow/Service/UsersService.java:103]: please implement this comment on token

File: src/main/java/com/yellow/ordermanageryellow/controller/UserController.java [hunk 3]
```diff
@@ -1,24 +1,102 @@
...
+        } catch (NotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+        return ResponseEntity.status(HttpStatus.OK).body(id);
     }
 
+    @PutMapping()
+    public ResponseEntity updateCustomer(@RequestBody Users user, @RequestHeader String token) {
```
Comment (moriyahazan) [inline src/main/java/com/yellow/ordermanageryellow/controller/UserController.java:67]: please rename to updateuser

File: src/main/java/com/yellow/ordermanageryellow/controller/UserController.java [hunk 4]
```diff
@@ -1,24 +1,102 @@
...
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+        return ResponseEntity.status(HttpStatus.OK).body("update success " + user.getFullName());
+
+    }
+
+    @GetMapping("/{pageNumber}/{pageSize}")
+    public ResponseEntity<List<UserDTO>> getAllCustomers(@PathVariable int pageNumber, @PathVariable int pageSize, @RequestHeader String token) {
```
Comment (moriyahazan) [inline src/main/java/com/yellow/ordermanageryellow/controller/UserController.java:80]: rename to getUsers
pageSize should be constant in system

File: src/main/java/com/yellow/ordermanageryellow/Service/UsersService.java [hunk 5]
```diff
@@ -1,40 +1,109 @@
...
+        if (UserRepository.existsById(user.getId()))
+            return UserRepository.save(user);
+        else
+            throw new NotFoundException("user not found");
+    }
+
+    @SneakyThrows
+    public HashMap<String, String> getCustomerByNames(String prefix) {
+
+        //company and role will be taken from token
```
Comment (moriyahazan) [inline src/main/java/com/yellow/ordermanageryellow/Service/UsersService.java:91]: please implement this comment on token

**Review by moriyahazan — CHANGES_REQUESTED** [review]
Good Job

File: src/main/java/com/yellow/ordermanageryellow/controller/UserController.java (hunk 4 above)
Comment (EstherFrauvirt) [reply src/main/java/com/yellow/ordermanageryellow/controller/UserController.java:80]: this api get only customers (by roleId)
to change  it anyway?

Comment (moriyahazan) [reply src/main/java/com/yellow/ordermanageryellow/controller/UserController.java:80]: For users, there are 2 APIs
one - getUsers for "user management" screen with page number
second - getCustomerNames for new order popup

**Review by moriyahazan — APPROVED** [review]

**Pull request merged**